tests/Data/ProgrammesDb/EntityRepository/PromotionRepository/FindPromotionsByPidTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Tests\BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\CoreEntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use Tests\BBC\ProgrammesPagesService\AbstractDatabaseTest;
+
+/**
+ * @covers BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository::<public>
+ */
+class FindPromotionsByPidTest extends AbstractDatabaseTest
+{
+    private $promotionRepository;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->loadFixtures(['PromotionsFixture']);
+        $this->promotionRepository = $this->getRepository('ProgrammesPagesService:Promotion');
+    }
+
+    public function testOnlyActivePromotionsAreReceived()
+    {
+        $brandPid = new Pid('b010t19z');
+        $dbPromotions = $this->promotionRepository->findActivePromotionsByPid($brandPid, 300, 0);
+
+        $this->assertEquals(['p000000h', 'p000001h', 'p000004h'], array_column($dbPromotions, 'pid'));
```

Comment:
Hi Michael. I renamed the function to findOnlyActivePromotions. Promotions are normal + superpromotions.

Later we have another function which is findOnlyActiveSuperpromotions. Which only return superpromotions. We do this because brand page need to fetch both (promotions and superpromotions). But for instance, episodes under the brand need (just an example) superpromotions.